### DISCUSSION TITLE
support create sqlite3 database?

### DISCUSSION THREAD

**Discussion by Jzhenli** [beeware/briefcase#1948] [Q&A] [answered]
Hi, I build android apk with briefcase, and in the code will create sqlite3 database, but when run the apk, it will fail as follows, so it doesn't support create database yet?
![image](https://github.com/user-attachments/assets/490bd5d5-be43-40e9-88ea-c3124c0b3496)

**Comment by freakboy3742** [comment, marked as answer]
There should be no problem reading, writing or creating SQLite databases - provided you name a file location that is writable. 

On Android, the current working directory isn't writable; if you specify a filename without a fully qualified path, it will be interpreted relative to the current working directory, which will cause errors.

If you're using Toga, the app object has a `paths` attribute that contains a collection of base paths you can use to build a writable filename (eg `self.paths.data / "my.db"` in your startup method will describe a database file in the user's "data" folder. You may need to create that folder first with mkdir; but once it exists, the file will be writable.

**Reply by Jzhenli** [reply]
Besides the self.paths.data, if there're some other ways to know where can create  folder for read/write data? you said some fully qualified path means which path?

**Reply by mhsmith** [reply]
"Fully qualified" just means it starts with a `/`, or a drive letter on Windows. `self.paths.data` is the only location that you're guaranteed to be able to write to on Android, and it will always be fully qualified.
